lib/settings_validation_wrapper.rb

```diff
@@ -0,0 +1,250 @@
+class SettingsValidationWrapper
+  include ActiveModel::Validations
+
+  class AbsoluteFilepathValidator < ActiveModel::EachValidator
+    def validate_each(record, attribute, value)
+      unless value.start_with('/')
+        record.errors.add attribute, 'is not an absolute path'
+      end
+    end
+  end
+
+  class ClassValidator < ActiveModel::EachValidator
+    def validate_each(record, attribute, value)
+      unless options[:in].include?(value.class)
+        record.errors.add attribute,
+          "must have value of one of the classes: #{options[:in]}"
+      end
+    end
+  end
+
+  class DirectoryValidator < ActiveModel::EachValidator
+    def validate_each(record, attribute, value)
+      unless File.directory?(value)
+        record.errors.add attribute, 'is not a directory'
+      end
+    end
+  end
+
+  class ElementsAreAbsoluteFilepathsValidator < ActiveModel::EachValidator
+    def validate_each(record, attribute, value)
+      valid = value.is_a?(Array) && value.all? do |elem|
+        %w(/ ~/).any? { |start| elem.start_with?(start) }
+      end
+      unless valid
+        record.errors.add attribute,
+          'all elements must be absolute filepaths'
+      end
+    end
+  end
+
+  class ElementsArePresentValidator < ActiveModel::EachValidator
+    def validate_each(record, attribute, value)
+      unless value.is_a?(Array) && value.all? { |elem| elem.present? }
+        record.errors.add attribute,
+          'all elements must not be blank'
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.